crates/db_schema/src/utils.rs

```diff
@@ -256,8 +257,10 @@ pub fn limit_and_offset(
 ) -> Result<(i64, i64), diesel::result::Error> {
   let page = match page {
     Some(page) => {
-      if page < 1 {
-        return Err(QueryBuilderError("Page is < 1".into()));
+      if !(1..=PAGE_LIMIT_MAX).contains(&page) {
+        return Err(QueryBuilderError(
+          format!("Page limit is > {PAGE_LIMIT_MAX}").into(),
```

Comment:
Should return a proper `LemmyError` if at all possible.

Reply:
This function returns a diesel error, and its just matching the other one, so I believe this is converted to a lemmyerror somewhere else.

In main we've already properly switched to lemmy errors.

Reply:
I mean it shouldnt be so hard to convert to LemmyError so that it returns a proper API error for clients.

Reply:
It would take a lot of work, (that's already done in main) as all of these DB functions, including the views here, return diesel errors. If someone wants to do that for `0.19` they can go ahead, but its really not worth it.